### PR TITLE
CoW protocol airdrop

### PR DESCRIPTION
**Pull request by jbrit**
## Checklist

- [ ] The PR modified the backend to include support for decoding CoW Protocol airdrops on Ethereum and gnosis

### CODE REVIEW

**Comment by jbrit** [comment]
sqldiff
```sql
INSERT INTO assets(rowid,identifier,name,type) VALUES(5386,'eip155:100/erc20:0xc20C9C13E853fc64d054b73fF21d3636B2d97eaB','CoW Protocol Virtual Token','C');
INSERT INTO common_asset_details(rowid,identifier,symbol,coingecko,cryptocompare,forked,started,swapped_for) VALUES(5386,'eip155:100/erc20:0xc20C9C13E853fc64d054b73fF21d3636B2d97eaB','vCOW','cow-protocol','COW',NULL,1644610920,NULL);
INSERT INTO evm_tokens(rowid,identifier,token_kind,chain,address,decimals,protocol) VALUES(4491,'eip155:100/erc20:0xc20C9C13E853fc64d054b73fF21d3636B2d97eaB','A',100,'0xc20C9C13E853fc64d054b73fF21d3636B2d97eaB',18,NULL);
```

File: rotkehlchen/tests/exchanges/test_woo.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def test_woo_assets_are_known(mock_woo):
 
 def test_query_online_trade_history_basic(mock_woo):
     """Assert that the expected arguments are passed to the `_api_query` method"""
-    start_ts, end_ts = (1634600000, 1634620000)
+    start_ts, end_ts = 1634600000, 1634620000
```
Comment (yabirgb) [inline rotkehlchen/tests/exchanges/test_woo.py:59]: this is not related to this change. State that in the description of the PR. Also why the test was passing before? That might indicate that something is wrong in the test logic?

File: rotkehlchen/constants/assets.py [hunk 2]
```diff
@@ -266,6 +266,7 @@
 A_PSP = Asset('eip155:1/erc20:0xcAfE001067cDEF266AfB7Eb5A286dCFD277f3dE5')
 A_SDL = Asset('eip155:1/erc20:0xf1Dc500FdE233A4055e25e5BbF516372BC4F6871')
 A_VCOW = Asset('eip155:1/erc20:0xD057B63f5E69CF1B929b356b579Cba08D7688048')
+A_VCOW_GNOSIS = Asset('eip155:100/erc20:0xc20C9C13E853fc64d054b73fF21d3636B2d97eaB')
 A_FPIS = Asset('eip155:1/erc20:0xc2544A32872A91F4A553b404C6950e89De901fdb')
```
Comment (yabirgb) [inline rotkehlchen/constants/assets.py:270]: by comparing with others like `A_POLYGON_POS_MATIC` and `A_OPTIMISM_ETH` we seem to have the chain before the asset name so it should be `A_GNOSIS_VCOW`

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        event.notes = notes
+        return True
+    return False
+
+
+def decode_cow_claim(
+        context: DecoderContext,
+        vcow: 'CryptoAsset',
+) -> DecodingOutput:
+    """Common decoder for CoW Protocol airdrops"""
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/airdrops.py:44]:
```suggestion
    """Common decoder for CoW Protocol airdrops in gnosis and ethereum"""
```

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        event.counterparty = counterparty
+        event.notes = notes
+        return True
+    return False
+
+
+def decode_cow_claim(
+        context: DecoderContext,
+        vcow: 'CryptoAsset',
+) -> DecodingOutput:
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/airdrops.py:43]: It fits in one line

```suggestion
def decode_cow_claim(context: DecoderContext, vcow: 'CryptoAsset') -> DecodingOutput:
```

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+COW_PROTOCOL_CLAIMED = b'\xd46\xe9\x97=\x1eD\xd4\r\xb4\xd4\x11\x9e<w<\xad\xb12;&9\x81\x96\x8c\x14\xd3\xd1\x91\xc0\xe1H'  # noqa: E501
+
+
+def make_claim(
+        event: 'EvmEvent',
+        user_address: 'ChecksumEvmAddress',
+        amount: 'FVal',
+        asset: 'Asset',
+        counterparty: str,
+        notes: Optional[str],
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/airdrops.py:28]: you already made it optional so if optional generate it using the template and if provided you can overwrite it

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 6]
```diff
@@ -57,6 +68,9 @@
 ELFI_LOCKING = string_to_evm_address('0x02Bd4A3b1b95b01F2Aa61655415A5d3EAAcaafdD')
 ELFI_VOTE_CHANGE = b'3\x16\x1c\xf2\xda(\xd7G\xbe\x9d\xf16\xb6\xf3r\x93\x90)\x84\x94\x94rht1\x93\xc5=s\xd3\xc2\xe0'  # noqa: E501
 
+COW_PROTOCOL = string_to_evm_address('0xD057B63f5E69CF1B929b356b579Cba08D7688048')
+COW_PROTOCOL_CLAIMED = b'\xd46\xe9\x97=\x1eD\xd4\r\xb4\xd4\x11\x9e<w<\xad\xb12;&9\x81\x96\x8c\x14\xd3\xd1\x91\xc0\xe1H'  # noqa: E501
+
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:73]: this here is not used in this file for what I'm seeing and is a duplicate. I believe you forgot to remove it

File: rotkehlchen/chain/gnosis/modules/airdrops/constants.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+from rotkehlchen.chain.ethereum.modules.airdrops.constants import CPT_COW
+from rotkehlchen.chain.gnosis.constants import CPT_GNOSIS
+
+
+GNOSIS_AIRDROPS_LIST = [
+    CPT_GNOSIS,
+    CPT_COW,
+]
```
Comment (yabirgb) [inline rotkehlchen/chain/gnosis/modules/airdrops/constants.py:8]:
```suggestion
GNOSIS_AIRDROPS_LIST = [CPT_GNOSIS, CPT_COW]
```

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 8]
```diff
@@ -183,16 +203,20 @@ def _decode_fpis_claim(
...
-                    continue
-                event.notes = f'Claim {amount} {crypto_asset.symbol} {note_location}'
+            notes = event.notes
+            try:
+                notes = f'Claim {amount} {event.asset.resolve_to_crypto_asset().symbol} {note_location}'  # noqa: E501
+            except (UnknownAsset, WrongAssetType):
+                self.notify_user(event=event, counterparty=counterparty)
+
+            if make_claim(
+                event,
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:213]: can you use the kwargs in all the arguments?

```suggestion
                event=event,
```

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 9]
```diff
@@ -81,21 +95,25 @@ def __init__(
...
         self.fpis = A_FPIS.resolve_to_evm_token()
         self.elfi = A_ELFI.resolve_to_evm_token()
+        self.vcow = A_VCOW.resolve_to_evm_token()
 
     def _decode_uniswap_claim(self, context: DecoderContext) -> DecodingOutput:
         if context.tx_log.topics[0] != UNISWAP_TOKEN_CLAIMED:
             return DEFAULT_DECODING_OUTPUT
 
-        user_address = hex_or_bytes_to_address(context.tx_log.data[32:64])
+        hex_or_bytes_to_address(context.tx_log.data[32:64])
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:104]: this here looks suspicous. Is a no-op. I believe the user address should be this instead

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 10]
```diff
@@ -104,16 +122,18 @@ def _decode_fox_claim(self, context: DecoderContext) -> DecodingOutput:
         if context.tx_log.topics[0] != FOX_CLAIMED:
             return DEFAULT_DECODING_OUTPUT
 
-        user_address = hex_or_bytes_to_address(context.tx_log.topics[1])
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:107]: keep it here to avoid computing it in every step of the for loop

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 11]
```diff
@@ -251,6 +275,7 @@ def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
             CONVEX: (self._decode_fpis_claim, 'convex'),
             FOX_DISTRIBUTOR: (self._decode_fox_claim,),
             ELFI_LOCKING: (self._decode_elfi_claim,),
+            COW_PROTOCOL: (lambda context: decode_cow_claim(context, self.vcow),),
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:278]: can you check if in gnosis and here it works to do

```suggestion
            ELFI_LOCKING: (self._decode_elfi_claim,),
            COW_PROTOCOL: (decode_cow_claim,  self.vcow),
```

if I recall correctly the first element is the decoding logic and then you have the args for the function excluding the context

**Review by yabirgb — COMMENTED** [review]
Some comments, shouldn't be much

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 12]
```diff
@@ -183,16 +203,20 @@ def _decode_fpis_claim(
             counterparty = CPT_FRAX
```
Comment (jbrit) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:203]: > Wouldn't it be possible to also generate the notes inside `make_claim`? It is always
> 
> ```
> Claim {amount} {asset.symbol} from {counterpary} airdrop
> ```

it is possible for most, there's this that's slightly different

File: rotkehlchen/tests/exchanges/test_woo.py (hunk 1 above)
Comment (jbrit) [reply rotkehlchen/tests/exchanges/test_woo.py:59]: it was a nitpick from a previous pr, both are valid

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 13]
```diff
@@ -251,6 +270,7 @@ def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
             CONVEX: (self._decode_fpis_claim, 'convex'),
             FOX_DISTRIBUTOR: (self._decode_fox_claim,),
             ELFI_LOCKING: (self._decode_elfi_claim,),
+            COW_PROTOCOL: (decode_cow_claim, self.vcow),
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:273]: This is the same as doing this
```suggestion
            self.vcow.evm_address: (decode_cow_claim, self.vcow),
```

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 14]
```diff
@@ -286,4 +306,9 @@ def counterparties(self) -> list[CounterpartyDetails]:
                 label='Element Finance',
                 image='element_finance.png',
             ),
+            CounterpartyDetails(
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:309]: What's this? There is no such thing. This is cowswap ...

File: rotkehlchen/chain/gnosis/modules/airdrops/decoder.py [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+from rotkehlchen.constants.assets import A_GNOSIS_VCOW
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.types import ChecksumEvmAddress
+    from rotkehlchen.user_messages import MessagesAggregator
+
+COW_PROTOCOL = string_to_evm_address('0xc20C9C13E853fc64d054b73fF21d3636B2d97eaB')
```
Comment (LefterisJP) [inline rotkehlchen/chain/gnosis/modules/airdrops/decoder.py:18]: again with this. You are repeating constants across file. And as I told you this is not needed. You have this address.
```suggestion
```

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+)
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.assets.asset import Asset, CryptoAsset
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.types import ChecksumEvmAddress
+
+COW_PROTOCOL_CLAIMED = b'\xd46\xe9\x97=\x1eD\xd4\r\xb4\xd4\x11\x9e<w<\xad\xb12;&9\x81\x96\x8c\x14\xd3\xd1\x91\xc0\xe1H'  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/airdrops.py:19]: Not used outside of here

```suggestion
CLAIMED = b'\xd46\xe9\x97=\x1eD\xd4\r\xb4\xd4\x11\x9e<w<\xad\xb12;&9\x81\x96\x8c\x14\xd3\xd1\x91\xc0\xe1H'  # noqa: E501
```

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+
+def make_claim(
+        event: 'EvmEvent',
+        user_address: 'ChecksumEvmAddress',
+        amount: 'FVal',
+        asset: 'Asset',
+        counterparty: str,
+        notes: Optional[str] = None,
+) -> bool:
+    """It matches an transfer event to an airdrop claim, changes the required fields
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/airdrops.py:30]:
```suggestion
    """It matches a transfer event to an airdrop claim, changes the required fields
```

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+     then returns `True` if a match was found"""
+    if not (event.event_type == HistoryEventType.RECEIVE and event.location_label == user_address and amount == event.balance.amount and asset == event.asset):  # noqa: E501
+        return False
+    event.event_subtype = HistoryEventSubType.AIRDROP
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/airdrops.py:34]: readability

```suggestion

    event.event_subtype = HistoryEventSubType.AIRDROP
```

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.assets.asset import Asset, CryptoAsset
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.types import ChecksumEvmAddress
+
+COW_PROTOCOL_CLAIMED = b'\xd46\xe9\x97=\x1eD\xd4\r\xb4\xd4\x11\x9e<w<\xad\xb12;&9\x81\x96\x8c\x14\xd3\xd1\x91\xc0\xe1H'  # noqa: E501
+
+
+def make_claim(
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/airdrops.py:22]: Why are you making a separate function that's used in only one place in the same file?

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+     then returns `True` if a match was found"""
+    if not (event.event_type == HistoryEventType.RECEIVE and event.location_label == user_address and amount == event.balance.amount and asset == event.asset):  # noqa: E501
+        return False
+    event.event_subtype = HistoryEventSubType.AIRDROP
+    event.counterparty = counterparty
+    event.notes = f'Claim {amount} {asset.symbol_or_name()} from {counterparty} airdrop' if notes is None else notes  # noqa: E501
+    return True
+
+
+def decode_cow_claim(context: DecoderContext, vcow: 'CryptoAsset') -> DecodingOutput:
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/airdrops.py:40]: What is this doing here? This is a very specific airdrop claim located in a file that should only contain generic code.

This file is in `/evm/decoding/airdrops.py` and should only have generic code.

Also why is a constant asset passed as an argument? :thinking: 

This should just be part of the decoder and use the `self.vcow`  ... that's not a place for it to be.

File: rotkehlchen/chain/gnosis/modules/airdrops/decoder.py [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.vcow = A_GNOSIS_VCOW.resolve_to_evm_token()
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict['ChecksumEvmAddress', tuple[Any, ...]]:
+        return {
+            COW_PROTOCOL: (decode_cow_claim, self.vcow),
```
Comment (LefterisJP) [inline rotkehlchen/chain/gnosis/modules/airdrops/decoder.py:43]:
```suggestion
            self.vcow.evm_address: (decode_cow_claim, self.vcow),
```

File: rotkehlchen/chain/gnosis/modules/airdrops/decoder.py [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+    def counterparties(self) -> list[CounterpartyDetails]:
+        return [CounterpartyDetails(identifier=CPT_COW, label='CoW Protocol', image='cow.svg')]
```
Comment (LefterisJP) [inline rotkehlchen/chain/gnosis/modules/airdrops/decoder.py:47]: same as before. There is no cow protocol.

File: rotkehlchen/data/global.db
Comment (LefterisJP) [inline rotkehlchen/data/global.db:1]: What's the global DB diff?

File: rotkehlchen/tests/unit/decoders/test_main.py [hunk 23]
```diff
@@ -100,6 +100,7 @@ def test_decoders_initialization(ethereum_transaction_decoder: 'EthereumTransact
         'makerdao vault',
         '1inch-v2',
         'uniswap',
+        'cow',
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/test_main.py:103]: not a protocol

File: rotkehlchen/data/global.db
Comment (yabirgb) [reply rotkehlchen/data/global.db:1]: The diff is here https://github.com/rotki/rotki/pull/6800#issuecomment-1775801855

File: rotkehlchen/chain/evm/decoding/airdrops.py (hunk 19 above)
Comment (jbrit) [reply rotkehlchen/chain/evm/decoding/airdrops.py:27]: it's used outside this file too

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 24]
```diff
@@ -0,0 +1,78 @@
...
+    if not (event.event_type == HistoryEventType.RECEIVE and event.location_label == user_address and amount == event.balance.amount and asset == event.asset):  # noqa: E501
+        return False
+
+    event.event_subtype = HistoryEventSubType.AIRDROP
+    event.counterparty = counterparty
+    event.notes = f'Claim {amount} {asset.symbol_or_name()} from {counterparty} airdrop' if notes is None else notes  # noqa: E501
+    return True
+
+
+class CommonAirdropsDecoder(DecoderInterface, metaclass=ABCMeta):
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/airdrops.py:46]: This has nothing in common with the rest. It's specific to cowswap

```suggestion
class CowswapAirdropDecoder(DecoderInterface, metaclass=ABCMeta):
```

File: rotkehlchen/chain/evm/decoding/airdrops.py (hunk 19 above)
Comment (LefterisJP) [reply rotkehlchen/chain/evm/decoding/airdrops.py:27]: If that's the came it needs a much better name
```suggestion
def match_airdrop_claim(
```

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 25]
```diff
@@ -251,6 +267,7 @@ def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
             CONVEX: (self._decode_fpis_claim, 'convex'),
             FOX_DISTRIBUTOR: (self._decode_fox_claim,),
             ELFI_LOCKING: (self._decode_elfi_claim,),
+            self.vcow.evm_address: (self._decode_cow_claim,),
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:270]: no need to do this. Can't you just move this to the `CowswapAirdropDecoder` and add it to this by calling super().addresses_to_decoders()?

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 26]
```diff
@@ -0,0 +1,78 @@
...
+    """Common decoder for cowswap airdrops in gnosis and ethereum"""
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/airdrops.py:47]: Also I would move this class to evm/decoding/cowswap/interfaces.py. This is not generic enough code to be here.

In that case also `CLAIMED` would also need to go there in the constants.

File: rotkehlchen/chain/gnosis/modules/airdrops/decoder.py [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+            self,
+            ethereum_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=ethereum_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+            vcow=A_GNOSIS_VCOW.resolve_to_evm_token(),
```
Comment (LefterisJP) [inline rotkehlchen/chain/gnosis/modules/airdrops/decoder.py:32]: pass it as an asset normally and do the `resolve_to_evm_token()` in the constructor.

File: rotkehlchen/chain/ethereum/modules/airdrops/decoder.py [hunk 28]
```diff
@@ -73,6 +83,7 @@ def __init__(
             evm_inquirer=ethereum_inquirer,
             base_tools=base_tools,
             msg_aggregator=msg_aggregator,
+            vcow=A_VCOW.resolve_to_evm_token(),
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/airdrops/decoder.py:86]: pass it as an asset normally and do the `resolve_to_evm_token()` in the constructor.

File: rotkehlchen/chain/ethereum/modules/airdrops/constants.py [hunk 29]
```diff
@@ -7,6 +7,7 @@
 CPT_FRAX = 'frax'
 CPT_SHAPESHIFT = 'shapeshift'
 CPT_ELEMENT_FINANCE = 'element-finance'
+CPT_COW = 'cow'
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/airdrops/constants.py:10]: Why did you mark this as resolved. This needs to go away from here.

File: rotkehlchen/chain/evm/decoding/airdrops.py [hunk 30]
```diff
@@ -0,0 +1,30 @@
...
+
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.assets.asset import Asset
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.types import ChecksumEvmAddress
+
+CLAIMED = b'\xd46\xe9\x97=\x1eD\xd4\r\xb4\xd4\x11\x9e<w<\xad\xb12;&9\x81\x96\x8c\x14\xd3\xd1\x91\xc0\xe1H'  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/airdrops.py:11]: This is not needed here as I wrote in the last review

**Review by LefterisJP — APPROVED** [review]
LGTM. Failing test is unrelated. 

`rotkehlchen/tests/unit/test_polygon_pos_inquirer.py::test_polygon_pos_nodes_prune_and_archive_status`
